src/java.base/share/classes/java/io/SimpleIO.java

```diff
@@ -0,0 +1,1037 @@
+     * followed by a line terminator.
+     * For example:
+     * {@snippet lang="java":
+     * var fruit = List.of("apple", "pear", "orange");
+     * printLines(fruit); // @highlight substring="printLines"
+     * }
+     * will print on the output stream as:
+     * {@snippet lang="text":
+     * apple
+     * pear
+     * orange
+     *
+     * }
+     *
+     * @param lines list of strings
+     * @throws NullPointerException if lines is null
+     */
+    public static void printLines(List<String> lines) {
+        Objects.requireNonNull(lines, "lines must not be null");
+        lines.forEach(System.out::println);
+    }
+
+    /**
+     * The contents of the file {@code filename} are read as a string.
+     * For example:
+     * {@snippet lang="java":
+     * var text = """
+     *    "Hope" is the thing with feathers
+     *    That perches in the soul
+     *    And sings the tune without the words
+     *    And never stops - at all
+     *    """;
+     * write("data.txt", text);
+     * var data = read("data.txt"); // @highlight substring="read"
+     * print(data);
+     * }
+     * will print on the output stream as:
+     * {@snippet lang="text":
+     * "Hope" is the thing with feathers
+     * That perches in the soul
+     * And sings the tune without the words
+     * And never stops - at all
+     *
+     * }
+     *
+     * @param filename  file name of file to be read
+     * @return Content read from a file
+     * @throws UncheckedIOException wrapping an IOException if an io error occurs.
+     *
+     * @implSpec Line terminators are normalized to '\n'.
+     */
+    public static String read(String filename) {
+        Objects.requireNonNull(filename, "filename must not be null");
+        return read(Path.of(filename));
+    }
+
+    /**
+     * The contents of the file at {@code path} are read as a string.
+     * For example:
+     * {@snippet lang="java":
+     * var text = """
+     *    "Hope" is the thing with feathers
+     *    That perches in the soul
+     *    And sings the tune without the words
+     *    And never stops - at all
+     *    """;
+     * var path = Path.of("data.txt");
+     * write(path, text);
+     * var data = read(path); // @highlight substring="read"
+     * print(data);
+     * }
+     * will print on the output stream as:
+     * {@snippet lang="text":
+     * "Hope" is the thing with feathers
+     * That perches in the soul
+     * And sings the tune without the words
+     * And never stops - at all
+     *
+     * }
+     *
+     * @param path  path of file to be read
+     * @return Content read from a file
+     * @throws UncheckedIOException wrapping an IOException if an io error occurs.
+     *
+     * @implSpec Line terminators are normalized to '\n'.
+     */
+    public static String read(Path path) {
+        Objects.requireNonNull(path, "path must not be null");
+        return readLines(path).stream()
+                .collect(Collectors.joining("\n", "", "\n"));
```

Comment:
See also Files.readString.

Reply:
Does `Files.readString` normalize the line terminators?

Reply:
The concern is to make sure students don't get surprised by platform terminators.